File: rb100/intro_ruby_round_two/basics/ex02.rb
# Launch School - Intro to Ruby
# Basics - ex02.rb
# Answered 07/07/2020

# Remaining Questions Explained: 
#   Expalin why this code works without having to declare 
#   variables first outside the loop.
# Resources: 
#   https://codingbee.net/ruby/ruby-variable-scope
#   https://launchschool.com/books/ruby/read/variables#variablescope

# thousands = 0
# hundreds = 0
# tens = 0
# ones = 0

while true do

    puts "Enter digit between 1 and 9999"
    num = gets.chomp().to_i

    if num.between?(1000,9999) then
        thousands = (num / 1000) % 10
        hundreds = (num / 100) % 10
        tens = (num / 10) % 10
        ones = (num / 1) % 10
        break   
    else 
        puts "You had one job! " +
            "Please re-enter number"
    end
end

puts "#{thousands} is is in the thousands place"
puts "#{hundreds} is is in the hundreds place"
puts "#{tens} is is in the tens place"
puts "#{ones} is is in the ones place."

File: js100/basics/ex02.js
/*
 * Launch School - Intro to Javascript
 * Basics - ex02.js
 * Answered 08/19/2020
*/

let num = 4936;


// let ones = num % 10;
// let tens = ((num % 100) - ones ) / 10;
// let hundreds = ((num % 1000) - (num % 100)) / 100;
// let thousands = ((num % 10000) - (num % 1000)) / 1000;

let thousands   = num % 10000;
let hundreds    = num % 1000;
let tens        = num % 100;
let ones        = num % 10;


thousands = (thousands - hundreds) / 1000;
hundreds  = (hundreds - tens) / 100;
tens      = (tens - ones) / 10;
// ones = ones;


console.log(num);
console.log(thousands);
console.log(hundreds);
console.log(tens);
console.log(ones);
